packages/valory/agents/oracle/tests/test_oracle.py

```diff
@@ -120,53 +119,66 @@ class TestABCIPriceEstimationSingleAgent(
     strict_check_strings = STRICT_CHECK_STRINGS
     happy_path = HAPPY_PATH
     package_registry_src_rel = PACKAGES_DIR
+    multisig = SERVICE_MULTISIG_1
+    key_pairs_override = KEY_PAIRS[:4]
+    _skill_name = PublicId.from_str(skill_package).name
+
+    @classmethod
+    def setup_class(cls) -> None:
+        """Set the class up."""
+        super().setup_class()
+        cls.extra_configs = [
+            {
+                "dotted_path": f"vendor.valory.skills.{cls._skill_name}.models.params.args.setup.safe_contract_address",
+                "value": json.dumps([cls.multisig]),
+                "type_": "list",
+            },
+        ]
+
+    def prepare_and_launch(self, nb_nodes: int) -> None:
+        """Prepare and launch the agents."""
+        # we need to set the correct key pairs, since the 5th agent is registered for the service with 1 agent instance.
+        self.key_pairs = self.key_pairs_override
+        super().prepare_and_launch(nb_nodes)
+
+
+@pytest.mark.e2e
+@pytest.mark.parametrize("nb_nodes", (1,))
+class TestABCIPriceEstimationSingleAgent(ABCIPriceEstimationTest):
+    """Test the ABCI oracle skill with only one agent."""
+
+    multisig = SERVICE_MULTISIG_2
+    key_pairs_override = [KEY_PAIRS[4]]
 
 
 @pytest.mark.e2e
 @pytest.mark.parametrize("nb_nodes", (2,))
-class TestABCIPriceEstimationTwoAgents(
-    BaseTestEnd2EndExecution,
-    UseGnosisSafeHardHatNet,
-):
+@pytest.mark.skip(
+    "Enable and set the correct multisig and key pairs "
+    "when registries image gets updated to include a service with 2 agent instances"
+)
```

Comment:
This is ready on https://github.com/valory-xyz/autonolas-registries/tree/deploy_service_two_agent_instances. I will create the image, push it, and then re-enable the e2e test on a separate PR.